src/propertySelector/propertySelector.js

```diff
@@ -0,0 +1,184 @@
+    return text.replace(match, highlightedTextTpl({ text: highlight }));
+}
+
+/**
+ * Creates property description list element
+ * @param {Object} property
+ * @returns JQuery element containing property description
+ */
+function createPropertyOption(property, search) {
+    const descriptionData = Object.assign({}, property);
+    if (search !== '') {
+        descriptionData.label = descriptionData.label && highlightCharacter(descriptionData.label, search, search);
+        descriptionData.alias = descriptionData.alias && highlightCharacter(descriptionData.alias, search, search);
+    }
+    const $propertyDescription = $(propertyDescriptionTpl({ property: descriptionData }));
+    const $checkboxContainer = $('.checkbox-container', $propertyDescription);
+    $checkboxContainer.append(checkBoxTpl({ id: descriptionData.id, checked: descriptionData.selected }));
+    return $propertyDescription;
+}
+
+export default function propertySelectorFactory(config) {
+    //element references
+    let $container;
+    let $buttonsContainer;
+    let $propertyListContaner;
+    let $searchInput;
+    let availableProperties = [];
+    let selectedProperties;
+    let search = '';
+
+    const searchRedrawTimeout = 500;
+    let searchRedrawTimeoutId;
+
+    const parentGap = 20;
+
+    const instance = component({
+        positionContainer() {
+            let { top, left, right, bottom } = this.config.data.position;
+            let maxHeight;
+            if (typeof bottom === 'undefined') {
+                maxHeight = $container.parent().height() - top - parentGap;
+            }
+            if (typeof top === 'undefined') {
+                maxHeight = $container.parent().height() - bottom - parentGap;
+            }
+            if (typeof top === 'undefined' && typeof bottom === 'undefined') {
+                top = 0;
+                bottom = 0;
+                maxHeight = $container.parent().height();
+            }
+
+            $container.css({ top, left, right, bottom, maxHeight });
+        },
+
+        /**
+         * Updates the list
+         */
+        redrawList() {
+            $propertyListContaner.empty();
+            const propertiesToRender = [];
+            availableProperties.forEach(property => {
+                property.selected = selectedProperties.has(property.id);
+                if (
+                    search === '' ||
+                    (property.label && property.label.toLowerCase().includes(search.toLowerCase())) ||
+                    (property.alias && property.alias.toLowerCase().includes(search.toLowerCase()))
+                ) {
+                    propertiesToRender.push(createPropertyOption(property, search));
+                }
+            });
+            $propertyListContaner.append(propertiesToRender);
+            this.trigger('redraw');
+        },
+        /**
+         * Setups search input event listners
+         */
+        setupSearch() {
```

Comment:
**thought:** This one should also move to the render step. You can still make it a dedicated function, but exposing is useless since it cannot be called by the consumer without harming the component as it will add the event listener each time.